### PR TITLE
Support for the Migration title and Source column is now implemented

### LINKED ISSUE #5631: Migrations UI list table should have Migration title and Source column
## User Story

As a site admin, I want the Migrations UI to have a _Migration title_ and _Source_ columns so that I can better understand what the source of the migration is and what it does.

## Details

Currently, there is no way to tell what migration does and what the source is. Having Migration title and Source columns should help users to have a better understanding of the Migration UI list table. 

## Acceptance Criteria

-   [x] Migration title column added to the Migration list table
-   [x] Source column added to the Migration list table

### PR DESCRIPTION
**Pull request by alaca**
Resolves #5631

## Description

This PR adds support for the Migration title and source columns. The `Migration` class is extended and 2 static methods, title and source, were added. 

## Affects

Migrations list table.
`Migration` class

## Visuals
image

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

Navigate to  `Donations -> Tools -> Data`

### CODE REVIEW

**Comment by alaca** [comment]
@JasonTheAdams this one is ready for your review.

**Review by JasonTheAdams — APPROVED** [review]
Great work! It really clears things up but works in a nicely backward-compatible way. 👍

File: src/Log/Migrations/CreateNewLogTable.php [hunk 1]
```diff
@@ -23,6 +23,13 @@ public static function id() {
 		return 'create_new_log_table';
 	}
 
+	/**
+	 * @return string
+	 */
+	public static function title() {
+		return  esc_html__( 'Create new give_log table' );
```
Comment (ravinderk) [inline src/Log/Migrations/CreateNewLogTable.php:30]: @alaca can we add text domain

File: src/Log/Migrations/DeleteOldLogTables.php [hunk 2]
```diff
@@ -17,6 +17,13 @@ public static function id() {
 		return 'delete_old_log_tables';
 	}
 
+	/**
+	 * @return string
+	 */
+	public static function title() {
+		return  esc_html__( 'Delete give_logs and give_logmeta tables' );
```
Comment (ravinderk) [inline src/Log/Migrations/DeleteOldLogTables.php:24]: @alaca can we add text domain

File: src/Log/Migrations/MigrateExistingLogs.php [hunk 3]
```diff
@@ -48,6 +48,13 @@ public static function id() {
 		return 'migrate_existing_logs';
 	}
 
+	/**
+	 * @return string
+	 */
+	public static function title() {
+		return  esc_html__( 'Migrate existing logs to give_log table' );
```
Comment (ravinderk) [inline src/Log/Migrations/MigrateExistingLogs.php:55]: @alaca can we add text domain

File: src/MigrationLog/Migrations/CreateMigrationsTable.php [hunk 4]
```diff
@@ -22,6 +22,13 @@ public static function id() {
 		return 'create_migrations_table';
 	}
 
+	/**
+	 * @return string
+	 */
+	public static function title() {
+		return  esc_html__( 'Create new give_migrations table' );
```
Comment (ravinderk) [inline src/MigrationLog/Migrations/CreateMigrationsTable.php:29]: @alaca can we add text domain

File: src/MigrationLog/Migrations/MigrateCompletedMigrations.php [hunk 5]
```diff
@@ -34,6 +34,13 @@ public static function id() {
 		return 'migrate_completed_migrations';
 	}
 
+	/**
+	 * @return string
+	 */
+	public static function title() {
+		return  esc_html__( 'Migrate completed migrations to give_migrations table' );
```
Comment (ravinderk) [inline src/MigrationLog/Migrations/MigrateCompletedMigrations.php:41]: @alaca can we add text domain

**Pull request merged**